edshipyard.py

```diff
@@ -69,92 +87,130 @@ def class_rating(module):
+                print('EDShipyard: {}'.format(e))
+
+            continue  # Silently skip unrecognized modules
+
+        except Exception:
+            if __debug__:
+                raise
 
     # Construct description
     ship = ship_map.get(data['ship']['name'].lower(), data['ship']['name'])
-    string = '[%s]\n' % (data['ship'].get('shipName') and ', '.join([ship, data['ship']['shipName']]) or ship)
-    for slot in ['H', 'L', 'M', 'S', 'U', None, 'BH', 'RB', 'TM', 'FH', 'EC', 'PC', 'SS', 'FS', None, 'MC', None, '9', '8', '7', '6', '5', '4', '3', '2', '1']:
+
+    if data['ship'].get('shipName') is not None:
+        _ships = '{}, {}'.format(ship, data['ship']['shipName'])
+    else:
+        _ships = ship
+
+    string = '[{}]\n'.format(_ships)
+
+    SLOT_TYPES = (
+        'H', 'L', 'M', 'S', 'U', None, 'BH', 'RB', 'TM', 'FH', 'EC', 'PC', 'SS', 'FS', None, 'MC', None, '9', '8',
+        '7', '6', '5', '4', '3', '2', '1'
+    )
+    for slot in SLOT_TYPES:
         if not slot:
             string += '\n'
+
         elif slot in loadout:
             for name in loadout[slot]:
-                string += '%s: %s\n' % (slot, name)
-    string += '---\nCargo : %d T\nFuel  : %d T\n' % (cargo, fuel)
+                string += '{}: {}\n'.format(slot, name)
+
+    string += '---\nCargo : {} T\nFuel  : {} T\n'.format(cargo, fuel)
 
     # Add mass and range
     assert data['ship']['name'].lower() in companion.ship_map, data['ship']['name']
     assert companion.ship_map[data['ship']['name'].lower()] in ships, companion.ship_map[data['ship']['name'].lower()]
+
     try:
+        # TODO: broken link
         # https://github.com/cmmcleod/coriolis/blob/master/app/js/shipyard/module-shipyard.js#L184
         mass += ships[companion.ship_map[data['ship']['name'].lower()]]['hullMass']
-        string += 'Mass  : %.2f T empty\n        %.2f T full\n' % (mass, mass + fuel + cargo)
+        string += 'Mass  : {:.2f} T empty\n        {:.2f} T full\n'.format(mass, mass + fuel + cargo)
+
         multiplier = pow(min(fuel, fsd['maxfuel']) / fsd['fuelmul'], 1.0 / fsd['fuelpower']) * fsd['optmass']
-        string += 'Range : %.2f LY unladen\n        %.2f LY laden\n' % (
+
+        string += 'Range : {:.2f} LY unladen\n        {:.2f} LY laden\n'.format(
             multiplier / (mass + fuel) + jumpboost,
-            multiplier / (mass + fuel + cargo) + jumpboost)
-    except:
-        if __debug__: raise
+            multiplier / (mass + fuel + cargo) + jumpboost
+        )
+
+    except Exception:
+        if __debug__:
+            raise
 
     if filename:
         with open(filename, 'wt') as h:
             h.write(string)
+
         return
 
     # Look for last ship of this type
     ship = companion.ship_file_name(data['ship'].get('shipName'), data['ship']['name'])
-    regexp = re.compile(re.escape(ship) + '\.\d\d\d\d\-\d\d\-\d\dT\d\d\.\d\d\.\d\d\.txt')
+    regexp = re.compile(re.escape(ship) + r'\.\d{4}-\d\d-\d\dT\d\d\.\d\d\.\d\d\.txt')
```

Comment:
You cited the more compact version in Discord, didn't push the whole change ?

Reply:
I felt that the further compacting did not actually make it any more compact and increased noise, `\d{2}` is five characters while `\d\d` is four, and easier to parse